### DISCUSSION TITLE
Blazor CSLA.Net 6.0.0 Error Synchronous operations are disallowed.

### DISCUSSION THREAD

**Discussion by jhuerta3** [MarimerLLC/csla#3016] [Questions] [answered]
Hello,

Im new to Blazor and im trying to get the Blazor sample working with EF. I am able to step in the code and confirm that I am getting data form the DB. The error 500: Internal Server Error System.InvalidOperationException: Synchronous operations are disallowed. 
Call WriteAsync or set AllowSynchronousIO to true instead.  I get the error on the public class PersonList : ReadOnlyListBase<PersonList, PersonInfo> and method 

        [Fetch]
        private void Fetch([Inject] DataAccess.IPersonDal dal, [Inject] IChildDataPortal<PersonInfo> personInfoPortal)
        {
            using (LoadListMode)
            {
                var data = dal.Get().Select(d => personInfoPortal.FetchChild(d));
                AddRange(data);
            }
        }

Razor code

@code {
  protected override async Task OnParametersSetAsync()
  {
     await vm.RefreshAsync(() => personListPortal.FetchAsync());
     var temp = vm.Exception;
  }
}

when I try to retrieve the data to the GUI. Is there something that I am missing? I tried to several things the last two days. I figured I ask just in case here in case there is a simple solution. In the meantime, I will continue to read up on Blazor and asynchronous. 

Thank you

**Comment by rockfordlhotka** [comment]
If you host the data portal using the http controller for HttpProxy, your server needs to be configured to allow sync IO. You can find examines of this in program.cs in the BlazorExample app server project.

**Comment by jhuerta3** [comment, marked as answer]
Got it. Sever. Program.css 

Added the code below and it worked. Thank you. 

// If using Kestrel:
builder.Services.Configure<KestrelServerOptions>(options =>
{
    options.AllowSynchronousIO = true;
});

// If using IIS:
builder.Services.Configure<IISServerOptions>(options =>
{
    options.AllowSynchronousIO = true;
});

**Reply by rockfordlhotka** [reply]
I'm glad you found it!

There is an item on the backlog to update the `MobileFormatter` to use the async APIs, thus avoiding this issue.
